Cargo.toml

```diff
@@ -100,6 +100,14 @@ nix = { version = "0.26.4", default-features = false, features = ["term"] }
 
 [build-dependencies]
 anyhow = "1.0.75"
+indexmap = { version = "2.1.0", features = ["serde"] }
```

Comment:
> Unfortunately I have quite a lot on my plate right now, so it's unclear when I will get the time to continue review.

No pressure! I'm in no rush.

> In general, the fewer dependencies the better, so I'm wondering if we need to depend on indexmap? Is it needed to make syntax mappings deterministic maybe?

Yes it is exactly as you stated. Without `IndexMap`, there would be no guarantee that a rule defined in front of another rule would be matched first, if their targets are different. For example:

```toml
[mappings]
"INI" = [ "foo0", "foo1" ]
"XML" = [ "bar0", "bar1" ]
```

Without `IndexMap`, it would still be guaranteed that `foo0` comes before `foo1` (because they are serialised as a `Vec`), but `bar0` and `bar1` may end up in front of `foo0` and `foo1`.